larpandoracontent/LArObjects/LArMCParticle.h

```diff
@@ -137,15 +178,57 @@ inline pandora::StatusCode LArMCParticleFactory::Read(Parameters &parameters, pa
     // ATTN: To receive this call-back must have already set file reader mc particle factory to this factory
     int nuanceCode(0);
 
+    int nMCStepPositions(0);
+    std::vector<pandora::InputCartesianVector> mcStepPositions;
+
+    int nMCStepMomentas(0);
+    std::vector<pandora::InputCartesianVector> mcStepMomentas;
+
     if (pandora::BINARY == fileReader.GetFileType())
     {
         pandora::BinaryFileReader &binaryFileReader(dynamic_cast<pandora::BinaryFileReader&>(fileReader));
         PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(nuanceCode));
+
+        PANDORA_RETURN_RESULT_IF(pandora::STATUS_CODE_SUCCESS, !=, binaryFileReader.ReadVariable(nMCStepPositions));
+        for (int i = 0; i < nMCStepPositions; ++i)
+        {
+            pandora::CartesianVector mcStepPosition(0.0,0.0,0.0);
```

Comment:
Purely cosmetic but you could probably rely on the default constructor here?

Reply:
No default constructor for `CartesianVector`s unfortunately.